lib/ansible/playbook/conditional.py

```diff
@@ -212,17 +213,32 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
             # to be looking for an undefined variable, return True, otherwise fail
             try:
                 # first we extract the variable name from the error message
-                var_name = re.compile(r"'(hostvars\[.+\]|[\w_]+)' is undefined").search(str(e)).groups()[0]
```

Comment:
micro optimization, we're now calling ```to_text(e)``` several times here.  Might as well do it once and store it into a variable.  Use the variable thereafter.